drafts/Stormless3.md

```diff
@@ -0,0 +1,322 @@
+StartAt: RecordDB
+States:
+  RecordDB:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-dev-RecordDB
+    Next: RecordAC
+    ResultPath: $.results.RecordDB
+  RecordAC:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-dev-RecordAC
+    Next: InviteSlack
+    ResultPath: $.results.RecordAC
+  InviteSlack:
+    Type: Task
+    Resource: arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-${self:custom.stage}-InviteSlack
+    ResultPath: $.results.InviteSlack
+    End: true
+
+```
+
+
+The StepFunction definition is written in [Amazon States Language](https://states-language.net/spec.html); the spec is short, well written and worth a read. Using YAML instead of JSON is a nice perk from the plugin - it reads better and allows comments. But if you want JSON - no problem, help yourself.
+
+`Resource` is used to refer the Lambda function by ARN. I used the variables we defined above to construct the ARSs matching account ID, region, and stage.
+
+`arn:aws:lambda:${self:custom.region}:${self:custom.private.accountId}:function:${self:service}-${self:custom.stage}-RecordDB`
+
+`ResultPath` is used to pass data between steps. By default, StepFunction works on the "need-to-know" basis: the step downstream receives the output from the step directly upstream. If you think it logical, think again: how does RecordAC and InviteSlack are gonna get contact's email? RecordDB may just return "200 OK". Forcing functions re-emit their input would introduce inappropriate intimacy. The trick is to use `ResultPaht` to append the output of each function under some key, like `ResultPath: $results.RecordDB`. Overriding `ResultPath` preserves initial workflow input for use from other Lambda steps. To fully grasp it, read the ResultPath section in the docs, or enjoy a ... < TODO: post from > 
+
+
+> PRO TIP: I made workflow sequential to demonstrate data passing. It is more proper to run all three things in parallel: it is faster and more resilient as a failure of one integration point will not stop workflow execution and the others will still be invoked. Go ahead change the StepFunctions to parallel.
+
+That is it. Time to try. Deploy, invoke, check the logs. 
+
+
+You CURL funs know what to do with the new API Gateway endpoint for our StepFunction. where exposed our I'll show off again with httpie, you CURL fanatics know 
+
+```
+# DON'T COPY! Use YOUR ENDPOINT!
+
+http POST https://YOUR.ENDPOINT.amazonaws.com/dev/signup \
+body:='{"email":"santa@mad.russian.xmas.com", "first_name":"Santa", "last_name": "Claus"}'
+```
+
+Ok, httpie or curl, it returns the StepFunction execution ARN so that we can check on it and see how the StepFunction is being executed. How? I afraid gotta open a browser and login to your AWS Console. First, try to use AWS CLI, don't say I didn't show you how:
```

Comment:
> How? I afraid gotta open a browser and login to your AWS Console.

This is a bit of a broken sentence - may want to remove it in lieu of the following sentence which shames them into CLI :)